apache2/ag_mdb/ag_mdb.cpp

```diff
@@ -660,36 +670,40 @@ int AGMDB_getExclusiveLock(struct agmdb_handler *dbm) {
 /**
  ** Free a shared lock that you have got before.
  ** @param dbm: the database you want to return the lock.
- ** return: AGMDB_SUCCESS if successfully created or AGMDB_FAIL if failed.
+ ** return: AGMDB_SUCCESS if successfully created or AGMDB_ERROR if failed.
  */
-int AGMDB_freeSharedLock(struct agmdb_handler *dbm){    
+int AGMDB_freeSharedLock(struct agmdb_handler *dbm){
+    int rc;
     struct agmdb_lock *db_lock;
     if(dbm == NULL)
-        return AGMDB_FAIL;
+        return AGMDB_ERROR_HANDLE_NULL;
     db_lock = (struct agmdb_lock *)&(dbm->db_lock);
     
-    if(Lock_V(db_lock, SEM_ID_READ, 1) == AGMDB_FAIL)
-        return AGMDB_FAIL;
+    rc = Lock_V(db_lock, SEM_ID_READ, 1);
+    if (rc != AGMDB_SUCCESS)
+        return rc;
 
     return AGMDB_SUCCESS;
 }
 
 /**
  ** Free a exclusive lock that you have got before.
  ** @param dbm: the database you want to return the lock.
- ** return: AGMDB_SUCCESS if successfully created or AGMDB_FAIL if failed.
+ ** return: AGMDB_SUCCESS if successfully created or AGMDB_ERROR if failed.
  */
 int AGMDB_freeExclusiveLock(struct agmdb_handler *dbm){
+    int rc;
     struct agmdb_lock *db_lock;
     if(dbm == NULL)
-        return AGMDB_FAIL;
+        return AGMDB_ERROR_HANDLE_NULL;
     db_lock = (struct agmdb_lock *)&(dbm->db_lock);
     
-    if(Lock_V(db_lock, SEM_ID_READ, SEM_READ_INITVAL) == AGMDB_FAIL)
-            return AGMDB_FAIL;
-        
-    if(Lock_V(db_lock, SEM_ID_WRITE, 1) == AGMDB_FAIL)
-        return AGMDB_FAIL;
+    rc = Lock_V(db_lock, SEM_ID_READ, SEM_READ_INITVAL);
+    if (rc != AGMDB_SUCCESS)
+        return AGMDB_ERROR;
```

Comment:
why don't we return the specific error code here intead of AGMDB_ERROR

Reply:
Will revise it